pycalphad/core/equilibrium.py

```diff
@@ -186,14 +186,7 @@ def equilibrium(dbf, comps, phases, conditions, output=None, model=None,
     if not broadcast:
         raise NotImplementedError('Broadcasting cannot yet be disabled')
     comps = sorted(unpack_components(dbf, comps))
-    phases = unpack_phases(phases) or sorted(dbf.phases.keys())
-    # remove phases that cannot be active
-    list_of_possible_phases = filter_phases(dbf, comps)
-    active_phases = sorted(set(list_of_possible_phases).intersection(set(phases)))
-    if len(list_of_possible_phases) == 0:
-        raise ConditionError('There are no phases in the Database that can be active with components {0}'.format(comps))
-    if len(active_phases) == 0:
-        raise ConditionError('None of the passed phases ({0}) are active. List of possible phases: {1}.'.format(phases, list_of_possible_phases))
```

Comment:
Coveralls report leads me to notice that this piece of code has become redundant after the last modifications in `filter_phases()`

Reply:
I suggest reverting this particular change, and changing it to call `filter_phases` twice: once with the `candidate_phases` keyword argument set to `None`, and the second time with `candidate_phases` set to `phases`. After the first call, check for `list_of_possible_phases` having length zero, as has been done, and then after the second call, you check `active_phases`.

This way, we can keep the `ConditionError` in `equilibrium` and `calculate`, where they have the most meaning, and `filter_phases` will not raise that exception when it might not make sense.